modules/core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/QueryParseException.java

```diff
@@ -21,6 +21,8 @@
 
 package org.nuxeo.ecm.core.query;
 
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
```

Comment:
What about using `javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST` like what we did in other Exception such as `DocumentNotFoundException` or `ConcurrentUpdateException`?

Same for other status code.

Reply:
I used the same SC_BAD_REQUEST as the PR [here](https://github.com/nuxeo/nuxeo/pull/4060/files)

Reply:
Oh well, ok ...
@efge shouldn't we want to align on `javax.servlet.http.HttpServletResponse` instead of relying on `org.apache.httpcomponents:httpcore` for those constants?

Reply:
@troger that would be a possible cleanup yes but I'd say the benefit is very low. We don't plan on removing HttpComponents from our dependencies anytime soon anyway.

Reply:
On the other hand for new code it's worth using standard JDK classes so `javax.servlet.http.HttpServletResponse` is slightly better

Reply:
@charlesboidot if you're not tired of doing iterations on this PR you could change the import yes, otherwise forget it

Reply:
It's ok I'll change it